packages/medusa/src/services/shipping-option.ts

```diff
@@ -397,6 +397,41 @@ class ShippingOptionService extends TransactionBaseService {
     return option
   }
 
+  private async validateAndMutatePrice(
+    option: ShippingOption,
```

Comment:
I agree with Adrien on this. I think we should think as functionally as possible and aim to avoid side-effects as much as possible. In addition to what Adrien said another bonus is readability, it's more explicit that something is re-assigned in the code.